engine/ingame/view.py

```diff
@@ -79,21 +81,28 @@ def _center_camera_to_player(self) -> None:
         v_width = self.camera.viewport_width
         v_height = self.camera.viewport_height
 
-        screen_center_x = min(
-            self.world_width - v_width,
-            max(
-                0,
-                player.center_x - v_width / 2,
-            ),
-        )
-        screen_center_y = min(
-            self.world_height - v_height,
-            max(
-                0,
-                player.center_y - v_height / 2,
-            ),
-        )
-
-        player_centered = screen_center_x, screen_center_y
-
-        self.camera.move_to(player_centered)
+        if self.world_width < v_width or self.world_height < v_height:
```

Comment:
So the reason why I requested to do each axis independently is that this code will break when one axis is long but the other is short.

Example: a map that is `(5000, 200)`. What this code will do is plop the camera right in the middle of that, but if the player moves to the left or to the right outside of the camera range, they will no longer be visible. In this situation the Y coordinate should be clamped at 100, but the X position will still be tied to player's X coordinate. The opposite situation would hold for a map with dimensions `(200, 5000)`.

Instead, the code would look something like this:
```python
if self.world_width < v_width:
  camera_x = middle of the screen
else:
  camera_x = tracking player X with the clamping at the edges

# similarly for Y

self.camera.move_to((camera_x, camera_y))
```

Reply:
Oooh, I see. OK!